File: problems/boj/4153.rs
fn main() {
  loop {
    let mut s = String::new();

    std::io::stdin().read_line(&mut s).unwrap();

    let v: Vec<i32> = s.split_whitespace().map(|s| s.parse().unwrap()).collect();
    let b: Vec<i32> = v.iter().map(|s| s.pow(2)).collect();

    let f: Vec<i32> = v.into_iter().filter(|s| *s == 0).collect();

    if f.len() == 3 {
      break;
    }

    if b[0] + b[1] == b[2] {
      println!("right");
    } else {
      println!("wrong");
    }
  }
}

File: problems/boj/4153.py
while True:
  a, b, c = map(int, input().split(' '))

  if a == 0 and b == 0 and c == 0:
    break

  ma = max([a, b, c])
  mi = min([a, b, c])
  mid = list(filter(lambda x: not x == ma and not x == mi, [a, b, c]))[0]

  print('right' if (mi**2) + (mid**2) == ma**2 else 'wrong')
